enzyme/Enzyme/MLIR/Interfaces/GradientUtilsReverse.h

```diff
@@ -79,6 +81,13 @@ class MGradientUtilsReverse {
   Type getIndexType();
   Value insertInit(Type t);
 
+  SmallVector<std::function<std::pair<Value, Value>(Type)>> cacheCreatorHook;
+  void
+  registerCacheCreatorHook(std::function<std::pair<Value, Value>(Type)> hook);
+  void
+  deregisterCacheCreatorHook(std::function<std::pair<Value, Value>(Type)> hook);
+  std::pair<Value, Value> getNewCache(Type t);
```

Comment:
It may make sense to template the closure parameter in order to avoid creating a `std::function` object. `template <typename Fun>`